dds/DCPS/DataReaderImpl.h

```diff
@@ -594,6 +599,55 @@ class OpenDDS_Dcps_Export DataReaderImpl
 
 protected:
 
+  // Update max flag if the spec ever changes
+  static const CORBA::ULong MAX_SAMPLE_STATE_FLAG = DDS::NOT_READ_SAMPLE_STATE;
+  static const CORBA::ULong MAX_SAMPLE_STATE_MASK = (MAX_SAMPLE_STATE_FLAG << 1) - 1;
+  static const CORBA::ULong MAX_SAMPLE_STATE_BITS = 2u;
+
+  // Update max flag if the spec ever changes
+  static const CORBA::ULong MAX_VIEW_STATE_FLAG = DDS::NOT_NEW_VIEW_STATE;
+  static const CORBA::ULong MAX_VIEW_STATE_MASK = (MAX_VIEW_STATE_FLAG << 1) - 1;
+  static const CORBA::ULong MAX_VIEW_STATE_BITS = 2u;
+
+  // Update max flag if the spec ever changes
+  static const CORBA::ULong MAX_INSTANCE_STATE_FLAG = DDS::NOT_ALIVE_NO_WRITERS_INSTANCE_STATE;
+  static const CORBA::ULong MAX_INSTANCE_STATE_MASK = (MAX_INSTANCE_STATE_FLAG << 1) - 1;
+  static const CORBA::ULong MAX_INSTANCE_STATE_BITS = 3u;
+
+  // These may need to be updated if the spec ever changes
+  static const CORBA::ULong COMBINED_VIEW_STATE_SHIFT = MAX_INSTANCE_STATE_BITS;
+  static const CORBA::ULong COMBINED_SAMPLE_STATE_SHIFT = COMBINED_VIEW_STATE_SHIFT + MAX_VIEW_STATE_BITS;
+  static const CORBA::ULong MAX_COMBINED_STATE_MASK = (MAX_SAMPLE_STATE_FLAG << (COMBINED_SAMPLE_STATE_SHIFT + MAX_SAMPLE_STATE_BITS)) - 1;
```

Comment:
Good catch. Since it's using `MAX_SAMPLE_STATE_FLAG` it doesn't need to factor in `MAX_SAMPLE_STATE_BITS`, so it's too large. But we don't end up using this constant anyway (I originally thought we'd need it) so I'll just remove it.